src/test/java/seedu/address/logic/parser/EditRecipeCommandParserTest.java

```diff
@@ -0,0 +1,290 @@
+        String recipeWithNegativeCalories =
+                CommandTestUtil.invalidRecipeCalorieField(CommandTestUtil.Number.NEGATIVE);
+        assertParseFailure(parser, recipeWithNegativeCalories, Calories.MESSAGE_CONSTRAINTS);
+
+
+        //The test cases testing for one invalid field
+        // invalid ingredient specified
+        assertParseFailure(parser, RECIPE_INDEX + " " + INVALID_INGREDIENT_DESC,
+                Ingredient.MESSAGE_CONSTRAINTS);
+
+        // invalid tag specified
+        assertParseFailure(parser, RECIPE_INDEX + " " + INVALID_TAG_DESC,
+                Tag.MESSAGE_CONSTRAINTS);
+
+        // invalid recipe name specified
+        assertParseFailure(parser, RECIPE_INDEX + " " + INVALID_NAME_DESC,
+                Name.MESSAGE_CONSTRAINTS);
+
+        // invalid recipe calorie specified: zero
+        assertParseFailure(parser, RECIPE_INDEX + " " + ZERO_CALORIES_DESC,
+                Calories.MESSAGE_CONSTRAINTS);
+
+        // invalid recipe calorie specified: non integer
+        assertParseFailure(parser, RECIPE_INDEX + " " + NON_INTEGER_CALORIES_DESC,
+                Calories.MESSAGE_CONSTRAINTS);
+
+        // invalid recipe calorie specified: negative
+        assertParseFailure(parser, RECIPE_INDEX + " " + NEGATIVE_CALORIES_DESC,
+                Calories.MESSAGE_CONSTRAINTS);
+
+
+        // while parsing {@code PREFIX_TAG} alone will reset the tags
+        // of the {@code Recipe} being edited,
+        // parsing it together with a invalid tag results in error
+        assertParseFailure(parser, RECIPE_INDEX + TAG_DESC_MARGARITAS
+            + TAG_DESC_NOODLE + INVALID_TAG_DESC, Tag.MESSAGE_CONSTRAINTS);
+        assertParseFailure(parser, RECIPE_INDEX + TAG_DESC_MARGARITAS
+            + TAG_DESC_NOODLE + MISSING_TAG_DESC, Tag.MESSAGE_CONSTRAINTS);
+        assertParseFailure(parser, RECIPE_INDEX + INVALID_TAG_DESC
+            + TAG_DESC_NOODLE + TAG_DESC_MARGARITAS, Tag.MESSAGE_CONSTRAINTS);
+        assertParseFailure(parser, RECIPE_INDEX + TAG_DESC_MARGARITAS
+            + MISSING_TAG_DESC + TAG_DESC_NOODLE, Tag.MESSAGE_CONSTRAINTS);
+
+
+        // multiple invalid values, but only the invalid value captured is ordered
+        // take note that missing instructions in edit command is valid
+        // by: 1. Name 2. Ingredient 3. Calories 4. Instructions 5. Recipe Image 6. Tags
+        assertParseFailure(parser, RECIPE_INDEX + INVALID_NAME_DESC + INVALID_INGREDIENT_DESC,
+                Name.MESSAGE_CONSTRAINTS);
+        assertParseFailure(parser, RECIPE_INDEX + NEGATIVE_CALORIES_DESC + INVALID_INGREDIENT_DESC,
+                Ingredient.MESSAGE_CONSTRAINTS);
+        assertParseFailure(parser, RECIPE_INDEX + MISSING_INSTRUCTIONS_DESC + NEGATIVE_CALORIES_DESC,
+                Calories.MESSAGE_CONSTRAINTS);
+        assertParseFailure(parser, RECIPE_INDEX + INVALID_TAG_DESC + MISSING_INSTRUCTIONS_DESC,
+               Tag.MESSAGE_CONSTRAINTS);
+    }
+
+    @Test
```

Comment:
Same as previous tests

Reply:
Will do these and more in next iteration.